### ISSUE TITLE
[BUG] Changing the coefficients makes `TrotterProduct` fail

### ISSUE THREAD

**Issue by minhtriet** [PennyLaneAI/pennylane#5066]
### Expected behavior

Both of the following snippets should work
(1)
```python
coeffs = [4, 5, 5, 5]
ops = [ qml.PauliZ(0) @ qml.PauliZ(1),  qml.PauliZ(0) @ qml.PauliZ(2),
          qml.PauliZ(0) @ qml.PauliZ(3), qml.PauliZ(0) @ qml.PauliZ(4)
      ]
H = qml.dot(coeffs, ops)

qml.TrotterProduct(H, time=2.4)
```
(2)
```python
coeffs= [5,5,5,5]
hamiltonian = qml.dot(coeffs, ops)

qml.TrotterProduct(hamiltonian,2.4)
```

### Actual behavior

(1) works but (2) doesn't

### System information

```shell
Name: PennyLane                                                               
Version: 0.34.0.dev0

Platform info:           Windows-10-10.0.19045-SP0
Python version:          3.10.9
Numpy version:           1.26.2
Scipy version:           1.11.4
Installed devices:
- default.gaussian (PennyLane-0.34.0.dev0)
- default.mixed (PennyLane-0.34.0.dev0)
- default.qubit (PennyLane-0.34.0.dev0)
- default.qubit.autograd (PennyLane-0.34.0.dev0)
- default.qubit.jax (PennyLane-0.34.0.dev0)
- default.qubit.legacy (PennyLane-0.34.0.dev0)
- default.qubit.tf (PennyLane-0.34.0.dev0)
- default.qubit.torch (PennyLane-0.34.0.dev0)
- default.qutrit (PennyLane-0.34.0.dev0)
- null.qubit (PennyLane-0.34.0.dev0)
- lightning.qubit (PennyLane-Lightning-0.33.1)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by trbromley** [comment]
Hi @minhtriet! Thanks for posting this bug.

We have a fix in #5073. This should be available in the 0.35 version of PennyLane released at the start of March. If you would like to use the fix sooner, you can wait until #5073 is merged and then run:
```python
pip uninstall pennylane -y
pip install git+https://github.com/PennyLaneAI/pennylane.git#egg=pennylane
```
This will install the development branch of PennyLane, so be prepared that there may be some rough edges!

**Comment by trbromley** [comment]
Closing this issue as it is fixed in #5073. @minhtriet let us know if you need any more help!
